Проектная работа №2

ValentinPside opened the pull request:
Первая попытка, вроде всё работает

ValentinPside committed:
Проектная работа №2

ilshat-abdulin commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java:
```diff
@@ -0,0 +1,26 @@
+
+public class BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T>{
+
+    private T t;
+
```
Это поле здесь никак не используется. Можно удалить.

ilshat-abdulin commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/FilterContactTypeAdapter.java:
```diff
@@ -103,5 +104,5 @@ public boolean areContentsTheSame(@NonNull FilterContactTypeUi oldItem, @NonNull
             return newItem;
         }
-    }
+    }*/
```
Это неиспользуемый ListDiffCallback можно удалить и так же поступить в остальных адаптерах

ilshat-abdulin commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/model/FilterContactTypeUi.java:
```diff
@@ -20,6 +23,11 @@ public boolean isSelected() {
         return selected;
     }
 
+    @Override
+    public boolean theSameAs(FilterContactTypeUi filterContactTypeUi) {
+        return this.hashCode() == filterContactTypeUi.hashCode();
+    }
+
```
⚠️ Обрати внимание, что вычисление должно происходить так, как это происходит в методе areItemsTheSame() класса ListDiffCallback в каждом адаптере.  Т.е. для FilterContactTypeUi здесь нужно сравнивать не хэшкоды, а его тип.

ilshat-abdulin commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/sort/SortTypeUI.java:
```diff
@@ -22,6 +23,11 @@ public boolean isSelected() {
         return selected;
     }
 
+    @Override
+    public boolean theSameAs(SortTypeUI sortTypeUI) {
+        return this.hashCode() == sortTypeUI.hashCode();
+    }
+
```
⚠️ Также, как в FilterContactTypeUI, не совсем то сравнивается

ValentinPside committed:
Проектная работа №2 (исправления)

Устранены критические замечания:
- в классе FilterContactTypeUi исправлена реализация метода theSameAs()
-  в классе SortTypeUI исправлена реализация метода theSameAs()
Дополнительно в адаптерах удалены неиспользуемые методы, которые ранее были закомментированы, а из класса  BaseListDiffCallback удалена неиспользуемая переменная.

ValentinPside commented:
Исправил замечания во втором комите
